src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractCreateServicesOnTextViewConnection.cs

```diff
@@ -47,18 +54,31 @@ void IWpfTextViewConnectionListener.SubjectBuffersDisconnected(IWpfTextView text
         /// <summary>
         /// Must be invoked from the UI thread.
         /// </summary>
-        private void CreateServices(string languageName)
+        private void CreateServicesOnUIThread()
         {
             var serviceTypeAssemblyQualifiedName = typeof(ISnippetInfoService).AssemblyQualifiedName;
             foreach (var languageService in _languageServices)
             {
                 if (languageService.Metadata.ServiceType == serviceTypeAssemblyQualifiedName &&
-                    languageService.Metadata.Language == languageName)
+                    languageService.Metadata.Language == _languageName)
                 {
                     _ = languageService.Value;
                     break;
                 }
             }
         }
+
+        private void CreateServicesInBackground()
+        {
+            _ = Task.Run(ImportCompeltionProviders);
+
+            void ImportCompeltionProviders()
```

Comment:
```suggestion
            // Preload completion providers on a background thread since assembly loads can be slow
            // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1242321
            void ImportCompletionProviders()
```